pkg/gridtypes/zos/vm.go

```diff
@@ -0,0 +1,98 @@
+package zos
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/zos/pkg/gridtypes"
+)
+
+// VirtualMachine reservation data
+type VirtualMachine struct {
+	Name     string   `json:"name"`
+	Network  string   `json:"network"`
+	IP       net.IP   `json:"ip"`
+	SSHKeys  []string `json:"ssh_keys"`
+	PublicIP string   `json:"public_ip"`
+	Size     uint8    `json:"size"`
+}
+
+// Valid implementation
+func (v VirtualMachine) Valid(getter gridtypes.WorkloadGetter) error {
```

Comment:
looks good, but I don't see that u actually use this in the Kubernetes struct. the Valid from that one can then use the VirtualMachine.Valid() then validates the extra k8s specific attributes